### ISSUE TITLE
Feature request: Indicator for selmode

### ISSUE THREAD

**Issue by dase78** [gokcehan/lf#1174]
I switch `selmode` quite regularly (great feature, bound to shortcuts!). Sometimes I'm not sure in which I am at the moment...

It would be really nice to have an indicator for the actual one used at the moment, a simple A or D at the bottom line right, just before the file-countings. This would fit to many other programs, take cmus for example.

**Comment by joelim-work** [comment]
The contents of the statusbar on the bottom right was previously fixed, but is now made configurable after #1168 

I haven't looked into that change in great detail, but I think it would be possible to modify the `ruler` option to support displaying `selmode` as well.

One concern I have is that because the statusbar is now configurable, this opens up a whole new set of possibilities of what to display. There may be other settings in `lf` that users wish to display, or other useful information like the amount of free disk space that was also introduced in #1168. And then there is the formatting and coloring of the statusbar components to consider. Currently this is not configurable - red is cut, yellow is copy, blue is filter, and magenta is select according to the code below, but there aren't many available terminal colors remaining. Perhaps it might be better to use format strings similar to the `promptfmt` option.

https://github.com/gokcehan/lf/blob/55ad40ac0ad11c979ed65e1a7406687fe9738d5c/ui.go#L830-L845

While I think it's fine for `selmode` to be added for your use case, I think the general idea a configurable statusbar merits further discussion.

**Comment by dase78** [comment]
I see, thanks a lot. Totally agree.

**Comment by joelim-work** [comment]
In the meantime, as a workaround you can type the following command to display the current value of `selmode`:

```
%printenv lf_selmode
```

This can be `map`ped to a particular key for convenience. Also, as a general tip you can use this technique to show the current value of any configuration option.

**Comment by dase78** [comment]
This works perfectly for me! And it opens a lot of configuring-fun... ;) Thanks!

**Comment by gokcehan** [comment]
@joelim-work That's a good point. If we start adding options to the statusbar then we might get too many similar requests. I did not mind `df` very much because ranger also has this in the statusbar. One could argue `selmode` is also special in a way that makes it an appropriate addition for the statusbar. Alternatively, maybe we could also consider a generic mechanism to display the the value of any option (e.g. `acc:progress:selection:ind:opt.selmode`) or maybe a generic mechanism to display the value of any environment variables so you could display the option value variables (e.g. `acc:progress:selection:ind:$lf_selmode`).

Configuration for select/copy/cut was discussed before in #754 though I'm not sure if there is a decision. Also I think the discussion there might predate the addition of `selmode` option.

**Comment by joelim-work** [comment]
It took me some time to think about how to implement this in a more generic way, but I have finally come up with something. I have described everything in #1255 which includes a demonstration.

**Comment by dase78** [comment]
This looks great! Hope it comes with r30.

**Comment by gokcehan** [comment]
@dase78 FYI, I have released r30 today as a bug fix release, so this might instead make it into another future release.

**Comment by dase78** [comment]
Thanx for the information! And for lf. :)
